File: csharp/test/Glacier2/sessionHelper/CallbackI.cs
// Copyright (c) ZeroC, Inc. All rights reserved.

using System.Threading;
using ZeroC.Ice;

namespace ZeroC.Glacier2.Test.SessionHelper
{
    public sealed class Callback : ICallback
    {
        public void InitiateCallback(ICallbackReceiverPrx? proxy, Current current, CancellationToken cancel) =>
            proxy!.Callback(current.Context, cancel: cancel);

        public void InitiateCallbackEx(ICallbackReceiverPrx? proxy, Current current, CancellationToken cancel) =>
            proxy!.CallbackEx(current.Context, cancel: cancel);

        public void Shutdown(Current current, CancellationToken cancel) =>
            current.Adapter.Communicator.ShutdownAsync();
    }
}

File: java/test/src/main/java/test/Glacier2/sessionHelper/CallbackI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package test.Glacier2.sessionHelper;

import test.Glacier2.sessionHelper.Test.CallbackException;
import test.Glacier2.sessionHelper.Test.CallbackReceiverPrx;
import test.Glacier2.sessionHelper.Test.Callback;

final class CallbackI implements Callback
{
    CallbackI()
    {
    }

    public void initiateCallback(CallbackReceiverPrx proxy, com.zeroc.Ice.Current current)
    {
        proxy.callback(current.ctx);
    }

    public void initiateCallbackEx(CallbackReceiverPrx proxy, com.zeroc.Ice.Current current)
        throws CallbackException
    {
        proxy.callbackEx(current.ctx);
    }

    public void shutdown(com.zeroc.Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }
}
